frontend/src/scenes/insights/insightLogic.ts

```diff
@@ -45,6 +74,77 @@ export const insightLogic = kea<insightLogicType>({
                 setAllFilters: (_, { filters }) => filters,
             },
         ],
+        /*
+        isFirstLoad determines if this is the first graph being shown after the component is mounted (used for analytics)
+        */
+        isFirstLoad: [
+            true,
+            {
+                setNotFirstLoad: () => false,
+            },
+        ],
+    }),
+    listeners: ({ values, actions }) => ({
+        setAllFilters: (filters) => {
+            actions.reportUsage(filters.filters)
+            actions.setNotFirstLoad()
+        },
+        reportUsage: async (filters: FilterType, breakpoint) => {
+            await breakpoint(500) // Debounce to avoid noisy events from changing filters multiple times
+
+            // Reports `insight viewed` event
+            const { display, interval, date_from, date_to, shown_as } = filters
+
+            // DEPRECATED: Remove when releasing `remove-shownas`
+            // Support for legacy `shown_as` property in a way that ensures standardized data reporting
+            let { insight } = filters
+            const SHOWN_AS_MAPPING: Record<string, 'TRENDS' | 'LIFECYCLE' | 'STICKINESS'> = {
+                Volume: 'TRENDS',
+                Lifecycle: 'LIFECYCLE',
+                Stickiness: 'STICKINESS',
+            }
+            if (shown_as) {
+                insight = SHOWN_AS_MAPPING[shown_as]
+            }
+
+            const properties: Record<string, any> = {
+                is_first_component_load: values.isFirstLoad,
+                insight,
+                display,
+                interval,
+                date_from,
+                date_to,
+                filters_count: filters.properties?.length || 0, // Only counts general filters (i.e. not per-event filters)
+                events_count: filters.events?.length, // Number of event lines in insights graph; number of steps in funnel
+                actions_count: filters.actions?.length, // Number of action lines in insights graph; number of steps in funnel
+            }
+
+            properties.total_event_actions_count = (properties.events_count || 0) + (properties.actions_count || 0)
+
+            // Custom properties for each insight
+            if (insight === 'TRENDS') {
```

Comment:
Not blocking: 

So this event is going to be hard to analyze because of properties sometimes being there and other times not.

On the other end, it's also a ton of code we now need to maintain and might start blowing up any time we change anything about filters leading to missing metrics.

Would we achieve the same with `properties.filters = filters`?

Reply:
As discussed on Slack, we'll keep both for now as having `filters` can future-proof us better, but having the parsed version enable us to do analytics **today** (with the disclaimer that this code may be a bit fragile).